docs/cadl-for-openapi-dev.md

```diff
@@ -0,0 +1,285 @@
+Cadl designs into OpenAPI.
+
+The document is organized around the features of an OpenAPI v2 or v3 definition.
+The idea is that if you know how to describe some API feature in OpenAPI, you can just navigate
+to the section of this document for that feature.
+
+## Data Types
+
+In OpenAPI [v2](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#data-types)/[v3](https://github.com/OAI/OpenAPI-Specification/blob/3.0.3/versions/3.0.3.md#dataTypes), data types are specified using the `type` and `format` fields in a schema.
+
+The Cadl equivalent of OpenAPI data types are the Cadl primitive types or [built-in models](https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#built-in-models).
+
+The following table shows how common OpenAPI types map to Cadl types:
+
+| OpenAPI `type`/`format`           | Cadl type        | Notes                  |
+| --------------------------------- | ---------------- | ---------------------- |
+| `type: integer, format: int32`    | `int32`          |                        |
+| `type: integer, format: int64`    | `int64`          |                        |
+| `type: number, format: float`     | `float32`        |                        |
+| `type: number, format: double`    | `float64`        |                        |
+| `type: string`                    | `string`         |                        |
+| `type: string, format: byte`      |                  |                        |
+| `type: string, format: binary`    | `byte[]`         | any sequence of octets |
+| `type: boolean`                   | `boolean`        |                        |
+| `type: string, format: date`      | `plainDate`      |                        |
+| `type: string, format: date-time` | `zonedDateTime`  | RFC 3339 date          |
+| `type: string, format: password`  | `@secret string` |                        |
+
+OpenAPI supports a variety of "assertions" that can be used further restrict the values allowed for a data type.
+These are actually borrowed into OpenAPI from JSON Schema.
+
+For `type: integer` and `type: number` data types:
+
+| OpenAPI/JSON Schema keyword | Cadl construct              | Notes |
+| --------------------------- | --------------------------- | ----- |
+| `minimum: value`            | `@minimum(value)` decorator |       |
+| `maximum: value`            | `@maximum(value)` decorator |       |
+
+For `type: string` data types:
+
+| OpenAPI/JSON Schema keyword | Cadl construct                | Notes |
+| --------------------------- | ----------------------------- | ----- |
+| `minLength: value`          | `@minLength(value)` decorator |       |
+| `maxLength: value`          | `@maxLength(value)` decorator |       |
+| `pattern: regex`            | `@format(regex)` decorator    |       |
+
+To define a `type: string` parameter or property with an `enum`, define the Cadl element as the union of the enum values.
+For example:
+
+```cadl
+status: "Running" | "Stopped" | "Failed"
```

Comment:
might want to describe the actual `enum` keyword too and when you should use which

Reply:
Agreed, enum would be a better representation than union of string